react-frontend/src/sharedComponents/GenerateRoutes.js

```diff
@@ -0,0 +1,38 @@
+import React from "react";
+import { Route } from "react-router-dom";
+import PropTypes from "prop-types";
+
+import PrivateRoute from "sharedComponents/PrivateRoute";
+
+function GenerateRoutes({ configs }) {
+  return (
+    <>
+      {configs.map((route) => {
```

Comment:
No inline functions within render/return.
Within the same file,  move it to a separate function (outside this component definition)